mne/viz/evoked.py

```diff
@@ -336,28 +336,34 @@ def _plot_lines(data, info, picks, fig, axes, spatial_colors, unit, units,
                                              exclude=[])
                     else:
                         layout = find_layout(info, None, exclude=[])
+                    head_pos = None
                     # drop channels that are not in the data
                     used_nm = np.array(_clean_names(info['ch_names']))[idx]
-                    names = np.asarray([name for name in used_nm
-                                        if name in layout.names])
-                    name_idx = [layout.names.index(name) for name in names]
+                    if layout is None:
+                        name_idx = list()
+                    else:
+                        names = np.asarray([name for name in used_nm
+                                            if name in layout.names])
+                        name_idx = [layout.names.index(name) for name in names]
                     if len(name_idx) < len(chs):
                         warn('Could not find layout for all the channels. '
                              'Generating custom layout from channel '
                              'positions.')
                         xy = _auto_topomap_coords(info, idx,
                                                   ignore_overlap=True)
                         layout = generate_2d_layout(
-                            xy, ch_names=list(used_nm), name='custom')
+                            xy, ch_names=list(used_nm), name='custom',
+                            normalize=False)
                         names = used_nm
                         name_idx = [layout.names.index(name) for name in
                                     names]
+                        head_pos = {'center': (0, 0), 'scale': (0.5, 0.5)}
```

Comment:
Well I think what would make sense here is for the user to be able to provide the coordinate frame. As part of the documentation some common frames like the MNI frame can be provided. Otherwise I guess there is nothing else you could anticipate on the code. WDYT?